sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java

```diff
@@ -115,6 +117,25 @@ protected BlobAsyncClientBase(HttpPipeline pipeline, String url, BlobServiceVers
         this.blobName = blobName;
         this.snapshot = snapshot;
         this.customerProvidedKey = customerProvidedKey;
+        this.blobPropertyHelper = null;
+        BlobHelper.PropertyAccessor propertyAccessor = new BlobHelper.PropertyAccessor() {
+            @Override
+            public BlobHelper getHelper(final BlobAsyncClientBase client) {
+                return client.blobPropertyHelper;
+            }
+
+            @Override
+            public AzureBlobStorageImpl getAzureBlobStorageImpl(final BlobAsyncClientBase client) {
+                return client.azureBlobStorage;
+            }
+
+            @Override
+            public CpkInfo getCustomerProvidedKey(final BlobAsyncClientBase client) {
+                return client.customerProvidedKey;
+            }
+        };
+
+        BlobHelper.setPropertyAccessor(propertyAccessor);
```

Comment:
What happens when multiple instances of `BlobAsyncClientBase` are created? This static `setPropertyAccessor()` method will update the reference of  `PropertyAccessor` each time a new instance of `BlobAsyncClientBase` is created. 

I guess it might work fine as the implementation is stateless today but it's something to consider and see if there are any side effects from this.